packages/web/src/pages/chat-page/components/ChatComposer.tsx

```diff
@@ -235,24 +156,21 @@ export const ChatComposer = (props: ChatComposerProps) => {
     }
     if (chatId !== chatIdRef.current) {
       setValue('')
-      setTrackId(null)
-      setCollectionId(null)
+      reset()
       chatIdRef.current = chatId
     }
-  }, [ref, chatId, chatIdRef, setTrackId, setValue])
+  }, [ref, chatId, chatIdRef, reset, setValue])
 
   const renderChatDisplay = (value: string) => {
-    const regexString = Object.keys(humanToData).join('|')
-    const regex = regexString ? new RegExp(regexString, 'gi') : null
-    if (!regex) {
+    const matches = getMatches(value)
+    if (!matches) {
       const text = splitOnNewline(value)
       return text.map((t, i) => (
         <ComposerText key={i} color='default'>
           {t}
         </ComposerText>
       ))
     }
-    const matches = value.matchAll(regex)
     const parts = []
     let lastIndex = 0
     for (const match of matches) {
```

Comment:
can this logic also be shared? or is it too diff between mobile/web?

Reply:
it's different. very tied to the text area set up. possibly a way but will need to think more